schema-registry/src/main/java/io/streamnative/pulsar/handlers/kop/schemaregistry/HttpJsonRequestProcessor.java

```diff
@@ -0,0 +1,125 @@
+    protected static final String RESPONSE_CONTENT_TYPE = "application/vnd.schemaregistry.v1+json";
+
+    private final Class<K> requestModel;
+    private final Pattern pattern;
+    private final String method;
+
+    public HttpJsonRequestProcessor(Class<K> requestModel, String uriPattern, String method) {
+        this.requestModel = requestModel;
+        this.pattern = Pattern.compile(uriPattern);
+        this.method = method;
+    }
+
+    protected static int getInt(int position, List<String> queryStringGroups) {
+        return Integer.parseInt(queryStringGroups.get(position));
+    }
+
+    protected static String getString(int position, List<String> queryStringGroups) {
+        return queryStringGroups.get(position);
+    }
+
+    @Override
+    public boolean acceptRequest(FullHttpRequest request) {
+        if (!request.method().name().equals(method)) {
+            return false;
+        }
+        List<String> groups = detectGroups(request);
+        if (groups == null) {
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public CompletableFuture<FullHttpResponse> processRequest(FullHttpRequest request) {
+        List<String> groups = detectGroups(request);
+        try (ByteBufInputStream inputStream = new ByteBufInputStream(request.content());) {
+            K decodeRequest;
+            if (requestModel == Void.class) {
+                decodeRequest = null;
+            } else {
+                decodeRequest = MAPPER.readValue((DataInput) inputStream, requestModel);
+            }
+            CompletableFuture<R> result;
+            try {
+                result = processRequest(decodeRequest, groups, request);
+            } catch (Exception err) {
+                result = FutureUtil.failedFuture(err);
+            }
+            return result.thenApply(resp -> {
+                if (resp == null) {
+                    return buildErrorResponse(NOT_FOUND, "Not found", "text/plain");
+                }
+                if (resp.getClass() == String.class) {
+                    return buildStringResponse(((String) resp), RESPONSE_CONTENT_TYPE);
+                } else {
+                    return buildJsonResponse(resp, RESPONSE_CONTENT_TYPE);
+                }
+            }).exceptionally(err -> {
+                log.error("Error while processing request", err);
+                return buildJsonErrorResponse(err);
+            });
+        } catch (Exception err) {
+            log.error("Error while processing request", err);
+            return CompletableFuture.completedFuture(buildErrorResponse(HttpResponseStatus.BAD_REQUEST,
+                    "Cannot decode request: " + err.getMessage(), "text/plain"));
+        }
+    }
+
+    private List<String> detectGroups(FullHttpRequest request) {
+        String uri = request.uri();
+        int questionMark = uri.lastIndexOf('?');
+        if (questionMark > 0) {
+            uri = uri.substring(0, questionMark);
+        }
```

Comment:
Could you add a `TODO` here since we just discard the query part of the URI. I see some REST API like`GET /schemas/ids/{int: id}` needs to process the query part.

> subject (string) – Add `?subject=<someSubjectName>` at the end of this request to look for the subject in all contexts starting with the default context, and return the schema with the id from that context.